### ISSUE TITLE
Issue when updating topics

### ISSUE THREAD

**Issue by LeoPerelli** [MaartenGr/BERTopic#622]
Hello and thanks for your great library!
I was running several runs with fixed UMAP hyper parameters to check the variability in the topic model. I sometimes get the following error (I stress "sometimes" because with same hyper parameters of UMAP but different run code sometimes works sometimes not, I got if from one of your answers to reduce topics)
![Screenshot 2022-07-19 at 18 23 52](https://user-images.githubusercontent.com/40518973/179801152-601b2b11-2389-4ef9-8bcf-590e7e3a666e.png)
In this case, 5 is the number of topics before the topic update call.
Any idea where the error might come from? I am running Bertopic with the latest version.
Thanks!

**Comment by drob-xx** [comment]
I'm sure @MaartenGr will know what is going on here. My guess is that you are winding up creating a UMAP model that HDBSCAN is creating less than five clusters for. BTW - my experience is that changes in UMAP params don't have a huge effect on cluster creation. It can, but often doesn't. However, HDBSCAN params will (usually) have a dramatic impact.

**Comment by MaartenGr** [comment]
As @drob-xx mentioned, there is a good chance that too few topics are created which might result in some issues here and there. Having said that, this typically should not happen. Based on some of the docstrings in your error message, it seems that you might not be using the latest version of BERTopic. Which version are you using?

**Comment by LeoPerelli** [comment]
Thank you for both the comments! Sorry, I didnt specify but the version is the latest, 0.11
By the way the strange thing is that the message is always "index x is out of bounds for axis 0 with size x", but x can also be quite big like 30, and indeed the topic model before updates has 30 topics.

**Comment by MaartenGr** [comment]
@LeoPerelli Could you share your entire code for training your model and getting to that specific error message? Also, are you using a public dataset for this? Perhaps I can recreate what is happening here.

**Comment by LeoPerelli** [comment]
@MaartenGr Thank you for your help. I am sharing the code, unfortunately I cannot share the data too. There is nothing special and I used this snippet countless times, however I was doing some hyper parameter tuning and apparently there are some combinations of parameters which lead to a UMAP dimensionality reduction which breaks something.

```count_vectorizer = CountVectorizer(min_df=8,ngram_range=(1, 3), stop_words=stopwords)
umap_model = UMAP(n_neighbors=20, n_components=50, min_dist=0.0, metric='cosine')
hdbscan_model = HDBSCAN(min_cluster_size=10,min_samples=1, metric='euclidean', cluster_selection_method='eom', prediction_data=True)

topic_model = BERTopic(vectorizer_model=count_vectorizer,umap_model=umap_model,hdbscan_model=hdbscan_model,calculate_probabilities=True,diversity=0.4)
topics, probs = topic_model.fit_transform(docs)
topics_arr = np.array(topics)

# Assign points in -1 cluster to most similiar cluster. 
probability_threshold = 0.01
new_topics = [np.argmax(prob) if max(prob) >= probability_threshold else -1 for prob in probs]
new_topics_arr = np.array(new_topics)

# Error comes from the following line:
topic_model.update_topics(docs, new_topics, n_gram_range=(1,3), vectorizer_model = count_vectorizer)

# Update topic frequencies
documents = pd.DataFrame({"Document": docs, "Topic": new_topics})
topic_model._update_topic_size(documents)
topics_after_update = topic_model.get_topic_info().sort_values(by='Count', ascending= False)```

**Comment by drob-xx** [comment]
I think that even though you are getting this error, that the model has been fitted. If that is the case I think that you can run this against your fitted BERTopic model which used the UMAP parameters which are giving you a problem:

`pd.Series(topic_model.hdbscan_model.labels_).value_counts()`

If that doesn't work then try:

`pd.Series(topic_model.hdbscan_model.fit_predict(topic_model.umap_model.embedding_)).value_counts()`

One of these should output a summary of the raw numbers of documents assigned to topic clusters which should reveal if there is a fundamental issue with the UMAP parameters you are using for your data.

**Comment by MaartenGr** [comment]
> I will try looking at what happens before and after reassigning the noise points, maybe one cluster gets emptied with this operation

I think this might be the case. The `update_topics` function is primarily created for updating the topic representations by using a different CountVectorizer. Later on, it has been used often for updating the actual points in a topic but seeing as that is not its intended purpose, you can have some issues if suddenly topics are missing. My guess would indeed be that a cluster is missing.

**Comment by MaartenGr** [comment]
Due to inactivity, I'll be closing this for now. Let me know if you have any other questions related to this and I'll make sure to re-open the issue!
